oneview/resource_server_profile.go

```diff
@@ -1416,19 +1454,14 @@ func resourceServerProfileCreate(d *schema.ResourceData, meta interface{}) error
 		for _, raw := range rawSanStorage {
 			sanStorageItem := raw.(map[string]interface{})
 			sanStorage = ov.SanStorageOptions{
-				HostOSType:            sanStorageItem["host_os_type"].(string),
-				ManageSanStorage:      sanStorageItem["manage_san_storage"].(bool),
-				ServerHardwareTypeURI: utils.NewNstring(sanStorageItem["server_hardware_type_uri"].(string)),
-				ServerHardwareURI:     utils.NewNstring(sanStorageItem["server_hardware_uri"].(string)),
-				SerialNumber:          sanStorageItem["serial_number"].(string),
-				Type:                  sanStorageItem["type"].(string),
-				URI:                   utils.NewNstring(sanStorageItem["uri"].(string)),
+				HostOSType:       sanStorageItem["host_os_type"].(string),
+				ManageSanStorage: sanStorageItem["manage_san_storage"].(bool),
 			}
 		}
 		serverProfile.SanStorage = sanStorage
 	}
 	if _, ok := d.GetOk("volume_attachments"); ok {
-		rawVolumeAttachments := d.Get("volume_attachments").(*schema.Set).List()
+		rawVolumeAttachments := d.Get("volume_attachments").([]interface{})
```

Comment:
what is the difference b/w ```interface``` and ```schema.Set```

Reply:
We change the  type from typeset to typelist so the conversion is added